### PR TITLE
Fix some Clang and Clazy warnings

### PR DESCRIPTION
**Pull request by gruenich**

### CODE REVIEW

**Comment by shundhammer** [comment]
Christoph, thank you for your contribution; but I am still in the process of cleaning up 15 years of technical debt left behind by a motley collection of people messing with the code without really understanding it.

I will gladly take your semicolon commit, but `QString::arg()` I really don't want to turn into less readable code with reintroducing over-long lines and an endless parameter list. I did that on purpose back then 20+ years ago, and I don't see a reason to change it.

As for unused variables - there are probably more, but gcc didn't warn about them. Some may be the result of commented-out code that had needed them. Some of that commented-out code is probably dead for good, but the person commenting it out didn't dare to simply delete it, so it was left over as comments that are often enough misleading. Some other commented-out code might be useful again for debugging.

That has to be decided on a case-by-case basis, and I am going through the code again and again to identify such cruft and kill it. But I don't want to throw a lot of stuff out immediately that might be needed after all; big changes make recovering such stuff more difficult.

**Comment by gruenich** [comment]
Thanks for providing feedback. I removed the `QString::args()` commit and only removed unused variables that are clearly left-overs and are not used for debugging or commented code.

The wrong variable name is clearly a bug, isn't it?

**Comment by shundhammer** [comment]
Not sure about that variable name; it might also just be re-using an existing temporary variable that was first used for the other version string. But whatever, it's clearer now, so it's really an improvement.

Thank you!

File: src/YQPkgPatchList.cc [hunk 1]
```diff
@@ -537,7 +537,7 @@ YQPkgPatchCategoryItem::patchCategory( const string & category )
 YQPkgPatchCategory
 YQPkgPatchCategoryItem::patchCategory( const QString & cat )
 {
-    QString category = category.toLower();
```
Comment (shundhammer) [inline src/YQPkgPatchList.cc:540]: This was a genuine bug that probably I introduced just yesterday: IIRC the parameter was a simple `QString`, and I turned it into a `const QString &`, and then gcc complained about that const thing being changed, so I had to rename the parameter, but I forgot to also change where that parameter was used. My bad.

**Pull request merged**